src/store/AddTag/useAddTag.ts

```diff
@@ -0,0 +1,29 @@
+import { useRecoilState } from 'recoil';
+
+import { TagType } from '~/components/common/Tag';
+import { useToast } from '~/store/Toast';
+
+import { addTagState } from './addTagStates';
+
+export function useAddTag() {
+  const [tags, setTags] = useRecoilState(addTagState);
+  const { fireToast } = useToast();
+
+  const hasTag = (tag: TagType) => {
+    return Boolean(tags.find(_tag => _tag.id === tag.id));
+  };
+
+  const addTag = (tag: TagType) => {
+    if (hasTag(tag)) {
+      fireToast({ content: '리스트에 테그가 이미 존재합니다.' });
+      return;
+    }
+    setTags([...tags, tag]);
```

Comment:
```suggestion
    setTags(prevTags => [...prevTags, tag]);
```

일반 `useState`의 setter처럼 이전 값을 보장해주는 방식으로 하면 좋을 거 같아요 !!!

Reply:
오 이거 중요해요!!! 저도 이것때매 여러번 삽질했답니다... ㅋㅋㅋㅋㅋㅋㅋ

React의 state는 기본적으로 비동기적으로 처리되서, 빠르게 처리되는 state에서 `[...prevTags]` 처럼 사용해버리는 경우 이전 state 값이 state 상태에 반영되지 않아서 값이 복제되더라고요!

혜성님께서 중요한 부분 집어주셨네요...!! 감사합니다

Reply:
어맛! 감사합니다! 오늘 또 React하나 배워갑니다!

Reply:
적용완료! d4a47b3